src/optimizer/rule/RemoveUselessProjectRule.cpp

```diff
@@ -0,0 +1,79 @@
+/* Copyright (c) 2021 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "optimizer/rule/RemoveUselessProjectRule.h"
+
+#include "optimizer/OptContext.h"
+#include "optimizer/OptGroup.h"
+#include "planner/PlanNode.h"
+#include "planner/Query.h"
+
+using nebula::graph::PlanNode;
+using nebula::graph::QueryContext;
+
+namespace nebula {
+namespace opt {
+
+std::unique_ptr<OptRule> RemoveUselessProjectRule::kInstance =
+    std::unique_ptr<RemoveUselessProjectRule>(new RemoveUselessProjectRule());
+
+RemoveUselessProjectRule::RemoveUselessProjectRule() {
```

Comment:
Prefer `PushProjectDownAggrRule` than this general meaningful name.

Reply:
Just delete unnecessary Project node here.  Maybe `Aggregate` is  just a special case.